src/main/java/org/damap/base/rest/invenio_damap/InvenioDAMAPResource.java

```diff
@@ -0,0 +1,178 @@
+      DmpService dmpService,
+      InvenioDAMAPService invenioDAMAPService) {
+    this.securityService = securityService;
+    this.accessValidator = accessValidator;
+    this.accessRepo = accessRepo;
+    this.dmpService = dmpService;
+    this.invenioDAMAPService = invenioDAMAPService;
+  }
+
+  SecurityService securityService;
+  AccessValidator accessValidator;
+  AccessRepo accessRepo;
+  DmpService dmpService;
+  InvenioDAMAPService invenioDAMAPService;
+
+  @ConfigProperty(name = "invenio.shared-secret")
+  String sharedSecret;
+
+  @Inject JWTParser parser;
+  @Inject
+  MapperService mapperService;
+
+  /*
+   * Maybe make it configurable?
+   * Could be more elaborate but should suffice for PoC
+   * Could also go into the {@link org.damap.base.security.SecurityService},
+   * but keeping everything together for PoC
+   *
+   */
+  private JsonWebToken validateAuthHeader(HttpHeaders headers) {
+    String jwtToken = headers.getHeaderString("X-Auth");
+
+    if (jwtToken != null && !jwtToken.isEmpty()) {
+      try {
+        JsonWebToken jwt = parser.verify(jwtToken, sharedSecret);
+
+        long exp = jwt.getExpirationTime();
+        long currentTime = System.currentTimeMillis() / 1000;
+
+        if (currentTime >= exp) throw new UnauthorizedException("Token expired.");
+
+        return jwt;
+      } catch (ParseException e) {
+        return null;
+      }
+    }
+    return null; // No Authorization header or not in the correct format
+  }
+
+  // Checks provided jwt.
+  private JsonWebToken checkIfUserIsAuthorized(HttpHeaders headers) {
+
+    JsonWebToken jwt = validateAuthHeader(headers);
+    boolean authorized = jwt != null;
+
+    if (!authorized) {
+      throw new UnauthorizedException("User unauthorized.");
+    }
+    return jwt;
+  }
+
+  // Resolves user identity based on the DMPs and returns dmp list and list of
+  // identifiers if successful.
+  private SimpleEntry<List<DmpDO>, List<String>> resolveDmpsAndIds(JsonObject identifiers) {
+    List<DmpDO> personDmpList = null;
+    List<String> matchingIdentifiers = new ArrayList<>();
+
+    for (String key : identifiers.keySet()) {
+      String identifier = identifiers.getString(key);
+      List<DmpDO> dmps = dmpService.getDmpDOListByPersonId(identifier);
+
+      // Check if returned DMPs is null or the list is empty.
+      // If yes, no resolving can take place, move to the next identifier.
+      if (dmps == null || dmps.isEmpty()) continue;
+
+      // Here dmps would contain at least one DMP so keep track of the current
+      // iterated identifier.
+      matchingIdentifiers.add(identifier);
+
+      // Keep track of the 1st list that matches at least one DMP.
+      // Then compare it with other non-null or non-empty lists.
+      if (personDmpList == null) personDmpList = dmps;
+
+      // If a DMP list is different than the previous one, a different identity was
+      // resolved.
+      else if (!personDmpList.equals(dmps))
```

Comment:
It is (theoretically) possible that two identifiers return the same dmps - but the ordering is not guaranteed by the db. Therefore, this compare should be made with containsAll(), just to be sure.